ui/raidboss/data/06-ew/raid/p9n.txt

```diff
@@ -1,6 +1,259 @@
+363.8 "Blizzard III" sync / 1[56]:[^:]*:Kokytos:(811F|8121):/
+363.8 "Fire III" sync / 1[56]:[^:]*:Kokytos:(811E|8120):/
+364.4 "--sync--" sync / 1[56]:[^:]*:Kokytos:811C:/
+370.5 "Global Spell" sync / 1[56]:[^:]*:Kokytos:8141:/
+371.4 "Global Spell" sync / 1[56]:[^:]*:Kokytos:8142:/
+372.7 "Disgorge" sync / 1[56]:[^:]*:Kokytos:8124:/
+376.8 "--sync--" sync / 1[56]:[^:]*:Kokytos:8115:/
+379.9 "--sync--" sync / 1[56]:[^:]*:Kokytos:8115:/
+
+
+
+# Fighter phase 2
+385.0 "Ravening (Fighter)" sync / 1[56]:[^:]*:Kokytos:8119:/
+395.1 "--sync--" sync / 1[56]:[^:]*:Kokytos:8127:/
+403.2 "Archaic Rockbreaker" sync / 1[56]:[^:]*:Kokytos:8128:/
+404.8 "Shockwave" sync / 1[56]:[^:]*:Kokytos:8129:/
+407.5 "--sync--" sync / 1[56]:[^:]*:Kokytos:8127:/
+410.6 "--sync--" sync / 1[56]:[^:]*:Kokytos:8127:/
+411.7 "Archaic Rockbreaker 1" sync / 1[56]:[^:]*:Kokytos:812A:/
+413.7 "--sync--" sync / 1[56]:[^:]*:Kokytos:8127:/
+416.7 "Archaic Rockbreaker 2" sync / 1[56]:[^:]*:Kokytos:812A:/
+421.8 "Front Combination/Rear Combination" sync / 1[56]:[^:]*:Kokytos:814[79]:/
+422.0 "Inside Roundhouse" sync / 1[56]:[^:]*:Kokytos:812B:/
+425.0 "Swinging Kick" sync / 1[56]:[^:]*:Kokytos:812[DE]:/
+433.1 "Gluttony's Augur" sync / 1[56]:[^:]*:Kokytos:8116:/
+433.5 "--sync--" sync / 1[56]:[^:]*:Kokytos:8117:/
+435.2 "--sync--" sync / 1[56]:[^:]*:Kokytos:8127:/
+438.3 "--sync--" sync / 1[56]:[^:]*:Kokytos:8127:/
+445.4 "Ascendant Fist" sync / 1[56]:[^:]*:Kokytos:8131:/
+447.5 "Disgorge" sync / 1[56]:[^:]*:Kokytos:8125:/
+451.6 "--sync--" sync / 1[56]:[^:]*:Kokytos:8115:/
+454.7 "--sync--" sync / 1[56]:[^:]*:Kokytos:8115:/
+457.9 "--sync--" sync / 1[56]:[^:]*:Kokytos:8144:/
+
+
+
+# Behemoth phase 2
+462.5 "Ravening (Behemoth)" sync / 1[56]:[^:]*:Kokytos:811A:/
+472.9 "--sync--" sync / 1[56]:[^:]*:Kokytos:8140:/
+476.0 "--sync--" sync / 1[56]:[^:]*:Kokytos:8140:/
+486.1 "Comet" sync / 1[56]:[^:]*:Kokytos:8134:/
+487.1 "--sync--" sync / 1[56]:[^:]*:Comet:8135:/
+495.1 "--sync--" sync / 1[56]:[^:]*:Kokytos:813B:/
+496.1 "--sync--" sync / 1[56]:[^:]*:Kokytos:813C:/
+497.1 "Ecliptic Meteor" sync / 1[56]:[^:]*:Kokytos:8188:/
+497.2 "--sync--" sync / 1[56]:[^:]*:Comet:8143:/
+501.2 "Beastly Bile" sync / 1[56]:[^:]*:Kokytos:813D:/
+503.1 "--sync--" sync / 1[56]:[^:]*:Kokytos:813E:/
+503.1 "Burst" sync / 1[56]:[^:]*:Comet:8136:/
+510.3 "Pulverizing Pounce" sync / 1[56]:[^:]*:Kokytos:813F:/
+513.6 "--sync--" sync / 1[56]:[^:]*:Kokytos:8140:/
+523.7 "Gluttonous Rampage" sync / 1[56]:[^:]*:Kokytos:8139:/
+523.9 "Ravenous Bite 1" #sync / 1[56]:[^:]*:Kokytos:813A:/
+526.1 "Ravenous Bite 2" #sync / 1[56]:[^:]*:Kokytos:813A:/
+528.1 "Touchdown" sync / 1[56]:[^:]*:Kokytos:8137:/
+528.5 "--sync--" sync / 1[56]:[^:]*:Kokytos:8146:/
+531.3 "Disgorge" sync / 1[56]:[^:]*:Kokytos:8126:/
+538.6 "--sync--" sync / 1[56]:[^:]*:Kokytos:8115:/
+541.7 "--sync--" sync / 1[56]:[^:]*:Kokytos:8115:/
+
+
+
+# Mage phase 3
+549.8 "Ravening (Mage)" sync / 1[56]:[^:]*:Kokytos:8118:/
```

Comment:
I am fairly sure this is the loop back to mage phase 2 based on my 11 minute log.  I see it go mage 2 and then fighter 2 up into gluttony's augur.  I think it's prooobably safe to assume a loop here.

Usually I look into fflogs in these cases and find the log with the longest time, but I have a hard time figuring out how to search by zones like this.  It might be there, but I'm not sure how to find it.

Reply:
If it's helpful, here's the lines from `make_timeline` from my log at this point.

```
544.8 "Ravening" sync / 1[56]:[^:]*:Kokytos:8118:/
553.9 "--sync--" sync / 1[56]:[^:]*:Kokytos:811C:/
557.0 "--sync--" sync / 1[56]:[^:]*:Kokytos:811C:/
559.1 "--sync--" sync / 1[56]:[^:]*:Kokytos:8144:/
565.5 "Iceflame Summoning" sync / 1[56]:[^:]*:Kokytos:86E1:/
570.4 "--sync--" sync / 1[56]:[^:]*:Kokytos:8122:/
570.7 "--sync--" sync / 1[56]:[^:]*:Kokytos:811C:/
573.8 "--sync--" sync / 1[56]:[^:]*:Kokytos:811C:/
576.4 "--sync--" sync / 1[56]:[^:]*:Ice Sphere:8729:/
576.4 "--sync--" sync / 1[56]:[^:]*:Fire Sphere:8728:/
576.4 "Sphere Shatter" sync / 1[56]:[^:]*:Ice Sphere:86E4:/
576.4 "Explosion" sync / 1[56]:[^:]*:Fire Sphere:86E3:/
581.0 "Iceflame Summoning" sync / 1[56]:[^:]*:Kokytos:86E1:/
585.9 "--sync--" sync / 1[56]:[^:]*:Kokytos:8123:/
588.3 "--sync--" sync / 1[56]:[^:]*:Kokytos:811C:/
591.9 "--sync--" sync / 1[56]:[^:]*:Ice Sphere:872A:/
591.9 "--sync--" sync / 1[56]:[^:]*:Fire Sphere:8727:/
591.9 "Sphere Shatter" sync / 1[56]:[^:]*:Ice Sphere:86E5:/
591.9 "Explosion" sync / 1[56]:[^:]*:Fire Sphere:86E2:/
596.5 "Dualspell" sync / 1[56]:[^:]*:Kokytos:811D:/
599.8 "--sync--" sync / 1[56]:[^:]*:Kokytos:8123:/
601.7 "--sync--" sync / 1[56]:[^:]*:Kokytos:811C:/
604.3 "Blizzard III" sync / 1[56]:[^:]*:Kokytos:8121:/
604.3 "Fire III" sync / 1[56]:[^:]*:Kokytos:811E:/
604.9 "--sync--" sync / 1[56]:[^:]*:Kokytos:811C:/
611.0 "Global Spell" sync / 1[56]:[^:]*:Kokytos:8141:/
611.8 "Global Spell" sync / 1[56]:[^:]*:Kokytos:8142:/
613.1 "Disgorge" sync / 1[56]:[^:]*:Kokytos:8124:/
617.1 "--sync--" sync / 1[56]:[^:]*:Kokytos:8115:/
620.2 "--sync--" sync / 1[56]:[^:]*:Kokytos:8115:/
625.3 "Ravening" sync / 1[56]:[^:]*:Kokytos:8119:/
635.4 "--sync--" sync / 1[56]:[^:]*:Kokytos:8127:/
643.5 "Archaic Rockbreaker" sync / 1[56]:[^:]*:Kokytos:8128:/
645.1 "Shockwave" sync / 1[56]:[^:]*:Kokytos:8129:/
647.8 "--sync--" sync / 1[56]:[^:]*:Kokytos:8127:/
650.9 "--sync--" sync / 1[56]:[^:]*:Kokytos:8127:/
652.0 "Archaic Rockbreaker" sync / 1[56]:[^:]*:Kokytos:812A:/
654.0 "--sync--" sync / 1[56]:[^:]*:Kokytos:8127:/
657.0 "Archaic Rockbreaker" sync / 1[56]:[^:]*:Kokytos:812A:/
662.1 "Rear Combination" sync / 1[56]:[^:]*:Kokytos:814A:/
662.3 "Outside Roundhouse" sync / 1[56]:[^:]*:Kokytos:812C:/
665.3 "Swinging Kick" sync / 1[56]:[^:]*:Kokytos:812E:/
673.4 "Gluttony's Augur" sync / 1[56]:[^:]*:Kokytos:8116:/
673.8 "Gluttony's Augur" sync / 1[56]:[^:]*:Kokytos:8117:/
```

Reply:
I found some FFLogs entries (e.g. [like this](https://www.fflogs.com/zone/reports?zone=54&boss=88&difficulty=0&class=Any&spec=Any&keystone=0&kills=0&duration=550)), but I wasn't able to find out how to download a network log from them to actually use those.

Reply:
That log you provided does line up more or less perfectly with a loop from Mage 2, so I think you're right.

Only exception is that I have Fighter 2 hardcoded as having an `Inside Roundhouse`, since in every single log I had, it's `Outside Roundhouse` in Fighter 1 and `Inside Roundhouse` in Fighter 2, so it seems to alternate and always be Outside on odds and Inside on evens.

I could either switch Fighter 2 to be `Inside/Outside Roundhouse` and loop, or I could fill out through Behemoth 3 and _then_ jump back to Mage 1, whichever you think is better.

Reply:
> I found some FFLogs entries (e.g. [like this](https://www.fflogs.com/zone/reports?zone=54&boss=88&difficulty=0&class=Any&spec=Any&keystone=0&kills=0&duration=550)), but I wasn't able to find out how to download a network log from them to actually use those.

You can do something like `node --loader ts-node/esm util/logtools/make_timeline.ts -k [YOUR KEY HERE] -r [FFLOGS REPORT ID] -rf [FIGHT NUMBER] -[OTHER ARGS]` to generate a timeline from FFLogs. It's not quite as nice as generating from a local network log, since it's missing some events, but it works for basic stuff.

Reply:
> Only exception is that I have Fighter 2 hardcoded as having an Inside Roundhouse, since in every single log I had, it's Outside Roundhouse in Fighter 1 and Inside Roundhouse in Fighter 2, so it seems to alternate and always be Outside on odds and Inside on evens.

If it's just one ability, it's probably better to just keep the timeline simple and have triggers cover the difference.  If'it's a branch with a whole bunch of abilities tied to it, then a branch in the timeline makes more sense.

> You can do something like node --loader ts-node/esm

I think you should just be able to `ts-node` directly these days without the loader args anymore.

Reply:
> Usually I look into fflogs in these cases and find the log with the longest time, but I have a hard time figuring out how to search by zones like this. It might be there, but I'm not sure how to find it.

I start [here](https://www.fflogs.com/zone/reports?zone=54&boss=88&difficulty=100&class=Any&spec=Any&kills=0&duration=0&server=0) and increase the "Minimum Pull Length" until I filter down to only a few logs.  From what I can see, [this](https://www.fflogs.com/reports/gC8AwyNma47pP92M#fight=1 ) ~28min log is the longest one so far.  Just have to be careful because sometimes the excessively long pulls are due to an ACT/FFLogs bug mashing two pulls together.

Reply:
Yeah, fflogs doesn't always have every ability or add.  I think(?) some abilties that can be dodged don't show up if nobody can be hit, but it definitely doesn't have things like head markers or game log lines, etc.  I tend to use fflogs only for testing, verifying loops, and finding enrage but "actual" network log files for making timelines.